primitives/src/communities.rs

```diff
@@ -229,20 +230,46 @@ impl Location {
 	}
 }
 
-#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, PartialOrd, Ord, TypeInfo)]
 #[cfg_attr(feature = "serde_derive", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "serde_derive", serde(rename_all = "camelCase"))]
-pub struct UnboundedCommunityMetadata {
-	/// utf8 encoded name
-	pub name: UnboundedPalletString,
-	/// utf8 encoded abbreviation of the name
-	pub symbol: UnboundedPalletString,
-	/// IPFS cid to assets necessary for community branding
-	pub assets: IpfsCid,
-	/// ipfs cid for style resources
-	pub theme: Option<IpfsCid>,
-	/// optional link to a community site
-	pub url: Option<UnboundedPalletString>,
+pub enum AnnouncementSigner {
+	Substrate(MultiSigner),
+	Bip340([u8; 32]),
+}
+
+impl Default for AnnouncementSigner {
+	fn default() -> Self {
+		AnnouncementSigner::Bip340([0u8; 32])
+	}
+}
+impl MaxEncodedLen for AnnouncementSigner {
+	fn max_encoded_len() -> usize {
+		34
+	}
+}
```

Comment:
It should not have problems deriving that.

Reply:
The problem is that MultiSigner doesn't implement MaxEncodedLen it seems

Reply:
Hmm, should we maybe just go for an account id anyhow, what is the benefit of having the multisigner?

Reply:
removed it altogether